src/soundio/soundmanagerconfig.cpp

```diff
@@ -393,6 +388,25 @@ unsigned int SoundManagerConfig::getAudioBufferSizeIndex() const {
 // This reflects the configured value only. In case of JACK the
 // setting of the JACK server is used.
 unsigned int SoundManagerConfig::getFramesPerBuffer() const {
+    if (m_api == MIXXX_PORTAUDIO_JACK_STRING) {
+        // in case of jack we configure the frames/period
+        if (m_audioBufferSizeIndex ==
+                static_cast<unsigned int>(
+                        JackAudioBufferSizeIndex::Size4096fpp)) {
+            return 4096;
+        } else if (m_audioBufferSizeIndex ==
+                static_cast<unsigned int>(
+                        JackAudioBufferSizeIndex::Size2048fpp)) {
+            return 2048;
+        }
+        // default is auto <= 1024
+        // The Jack buffer size can change at any time, so we
+        // need buffers for the maximum of 1024 (limited by Portaudio).
+        return 1024;
+    }
+
+    // With the other APIs we calc the frames per buffer form the sample rate
+
     // endless loop otherwise
     unsigned int audioBufferSizeIndex = m_audioBufferSizeIndex;
```

Comment:
```suggestion
    AudioBufferSizeIndex audioBufferSizeIndex = m_audioBufferSizeIndex;
```
and use the type for `m_audioBufferSizeIndex` too

Reply:
That does not work because it is used below in the calculation which requires the integer type. We may argue that we adjust all the code to achieve the desired type safety. But that will lead to a bigger refactoring that does not suite to a Bugfix PR in a stable branch. 

The underlying issue is that the indexes 1...7 are inherited from a legacy version of Mixxx and this code makes it compatible to the current version. The unsigned int is now actually a union type that represents different buffer sizes depending on the selected API an sample rate. This PR is on the middle ground that makes the new code readable according to your request but keeps the rest unchanged.